custom_components/kia_uvo/Vehicle.py

```diff
@@ -64,7 +65,16 @@ async def update(self):
                     self.set_geocoded_location, current_vehicle_location
                 )
 
-            async_dispatcher_send(self.hass, self.topic_update)
+            if (
+                self.get_child_value("vehicleStatus.engine") == False
+                and previous_vehicle_status.engine == False
+                and self.get_child_value("vehicleStatus.evStatus.batteryStatus") == 0
+                and previous_vehicle_status["evStatus"]["batteryStatus"] != 0
+            ):
+                await self.force_update()
```

Comment:
Let's fire this as a separate request rather than waiting results

Reply:
@fuatakgun I can switch this ti a second request, but the reason I did it with waiting is so that scripts in HA that call the update function are able to tell when it actually finishes.  If this is changed to async the scripts will call update thinking the values are updated but they won't be until the force update call finishes.

Reply:
ok, I see your point